eo-maven-plugin/src/test/java/org/eolang/maven/util/HmOptionalTest.java

```diff
@@ -0,0 +1,173 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.cactoos.text.Randomized;
+import org.cactoos.text.TextOf;
+import org.cactoos.text.UncheckedText;
+import org.eolang.maven.log.CaptureLogs;
+import org.eolang.maven.log.Logs;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+/**
+ * Test for {@link HmOptional}.
+ *
+ * @since 0.35.0
+ */
+final class HmOptionalTest {
+
+    /**
+     * Number of generated characters for testing.
+     */
+    private int size;
+
+    /**
+     * String sample for testing.
+     */
+    private String sample;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.size = 100;
+        this.sample = "file.txt";
+    }
+
+    @ParameterizedTest
+    @CsvSource({"true", "false"})
+    void savesIfFileNotExist(final boolean rewrite, @TempDir final Path dir) throws IOException {
+        final HmOptional optional = new HmOptional(new HmBase(dir), rewrite);
+        final Path file = Paths.get(this.sample);
+        final String content = new UncheckedText(new Randomized(this.size)).asString();
+        optional.save(content, file);
+        MatcherAssert.assertThat(
+            new UncheckedText(new TextOf(dir.resolve(file))).asString(),
+            Matchers.is(content)
+        );
+    }
+
+    @Test
+    void savesIfFileExistAndRewriteTrue(@TempDir final Path dir) throws IOException {
+        final String first = new UncheckedText(new Randomized(this.size)).asString();
+        final Path file = Paths.get(this.sample);
+        final HmBase base = new HmBase(dir);
+        base.save(first, file);
+        final HmOptional optional = new HmOptional(base, true);
+        final String second = new UncheckedText(new Randomized(this.size)).asString();
+        optional.save(second, file);
+        MatcherAssert.assertThat(
+            new UncheckedText(new TextOf(dir.resolve(file))).asString(),
+            Matchers.not(first)
+        );
+        MatcherAssert.assertThat(
+            new UncheckedText(new TextOf(dir.resolve(file))).asString(),
+            Matchers.is(second)
+        );
+    }
+
+    @Test
+    @CaptureLogs
+    void savesIfFileExistAndRewriteFalse(
```

Comment:
@Yanich96 I think the right english name for the test would be `savesIfFileExistsAndRewriteFalse`